webapp/src/sw.js

```diff
@@ -6,3 +6,28 @@ self.addEventListener('activate', (event) => {
   console.log('sw activated');
   event.waitUntil(self.clients.claim());
 });
+
+self.addEventListener('fetch', (event) => {
+  const { request } = event;
+
+  const response = caches
+    .match(request)
+    .then((response) => {
+      if (response) {
+        return response;
+      }
+
+      return fetch(request)
+        .then(resp => {
+          if (request.url.endsWith('.png') || request.url.endsWith('.jpg')) {
```

Comment:
wydaje mi się, ze powinniśmy dodać cache nie tylko do obrazków ale do całości assetów razem z js'em i html'em jeśli to ma działać jako PWA.